### PR TITLE
[CWS] add runtime security agent command to download policy

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR adds a command to the security agent runtime subcommand to download the CWS policy from the API.
It also reworks policy-related commands:
- `security-agent runtime check-policies -> security-agent runtime policy check`
- `security-agent runtime reload -> security-agent runtime policy reload`
- `security-agent runtime policy download`

Old commands are kept for now, but marked deprecated and will print a message when used

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/security-agent/app/runtime.go [hunk 1]
```diff
@@ -253,3 +276,68 @@ func startRuntimeSecurity(hostname string, stopper restart.Stopper, statsdClient
...
+func downloadPolicy(cmd *cobra.Command, args []string) error {
+	if !coreconfig.Datadog.IsSet("api_key") {
+		return errors.New("no API key is set")
+	}
+
+	if !coreconfig.Datadog.IsSet("app_key") {
+		return errors.New("no application key is set")
+	}
+
+	apiKey := coreconfig.Datadog.GetString("api_key")
```
Comment (lebauce) [inline cmd/security-agent/app/runtime.go:289]: What about simply checking that apiKey and appKey are not empty ?

File: cmd/security-agent/app/runtime.go [hunk 2]
```diff
@@ -253,3 +276,68 @@ func startRuntimeSecurity(hostname string, stopper restart.Stopper, statsdClient
...
+	appKey := coreconfig.Datadog.GetString("app_key")
+
+	site := coreconfig.Datadog.GetString("site")
+	if site == "" {
+		site = "datadoghq.com"
+	}
+
+	var outputPath string
+	if len(args) == 0 {
```
Comment (lebauce) [inline cmd/security-agent/app/runtime.go:298]: Using an optional option (--outputPath ?) provides a bit more flexiblity

File: cmd/security-agent/app/runtime.go (hunk 1 above)
Comment (paulcacheux) [reply cmd/security-agent/app/runtime.go:321]: It would be better yes

File: releasenotes/notes/cws-cmd-download-policy-0b19ce628afd289f.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    The security agent now offers a command to directly download the policy file from the API.
```
Comment (kayayarai) [inline releasenotes/notes/cws-cmd-download-policy-0b19ce628afd289f.yaml:11]:
```suggestion
    The security Agent now offers a command to directly download the policy file from the API.
```

File: releasenotes/notes/cws-cmd-download-policy-0b19ce628afd289f.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+deprecations:
+  - |
+    The commands `check-policies` and `reload` of the security agent are deprecated.
+    Please use `runtime policy check` and `runtime policy reload` respectively instead.
```
Comment (kayayarai) [inline releasenotes/notes/cws-cmd-download-policy-0b19ce628afd289f.yaml:15]:
```suggestion
    The security Agent commands `check-policies` and `reload` are deprecated.
    Use `runtime policy check` and `runtime policy reload` respectively instead.
```
Not sure, but these might require double back-ticks?

**Review by kayayarai — CHANGES_REQUESTED** [review]

File: releasenotes/notes/cws-cmd-download-policy-0b19ce628afd289f.yaml (hunk 4 above)
Comment (paulcacheux) [reply releasenotes/notes/cws-cmd-download-policy-0b19ce628afd289f.yaml]: Good point, let me check in other release notes

**Review by kayayarai — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

**Pull request merged**